packages/react-components/react-tag-picker-preview/src/components/TagPickerInput/useTagPickerInput.tsx

```diff
@@ -122,3 +148,25 @@ function useTagPickerContexts() {
     popoverId: useTagPickerContext_unstable(ctx => ctx.popoverId),
   };
 }
+
+/**
+ * while typing the user might need a bit more of space to see the text,
+ * which means the input should stretch to 100% width
+ * occupying a whole new line.
+ *
+ * This function will set the CSS variable `--width` to `100%` if the scrollWidth is greater than the offsetWidth,
+ * meaning the text is overflowing the input.
+ *
+ * @param input - input element to apply the style
+ * @returns void
+ */
+const setTagPickerInputStretchStyle = (input: HTMLInputElement) => {
+  // first we need to remove the CSS variable
+  // to properly calculate the difference between scrollWidth and offsetWidth
+  input.style.removeProperty(tagPickerInputCSSRules.width);
```

Comment:
In order to be overridable, I would suggest that you use this to set a CSS var and then use the var in normal class based styles